Platformer_001/Assets/MusicGenerator.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System;
+using UnityEngine;
+
+public class MusicGenerator : MonoBehaviour {
+
+	// Use this for initialization
+	void Start () {
+		DontDestroyOnLoad (transform.gameObject);
+		r = new System.Random();
+	}
+
+	public double frequency = 440;
+	public double gain = 1;
+
+	private double increment;
+	private double phase;
+	private double sampling_frequency = 48000;
+
+	public double ticksPerBar = 100;
+	public double ticks;
+	private bool noteChanged = true;
+
+	private double[] notes = {
+		261.63,
+		293.66,
+		329.63,
+		349.23,
+		392,
+		440,
+		493.88
+	};
+
+	private System.Random r;
+
+	void FixedUpdate() {
+		ticks += Time.deltaTime;
+		while (ticks > ticksPerBar) {
+			ticks -= ticksPerBar;
+			noteChanged = false;
+		}
+		if (ticks < 5 && !noteChanged) {
+			int nextIndex = r.Next (0, 6);
```

Comment:
Wir sind hier in Unity :stuck_out_tongue:! Da macht man `Random.Range(0, 6);`. Dann brauchst du auch `r` nicht mehr.